graylog2-web-interface/src/views/bindings.jsx

```diff
@@ -143,7 +144,13 @@ export default {
       defaultWidth: 4,
       visualizationComponent: AggregationBuilder,
       editComponent: AggregationControls,
-      needsControlledHeight: true,
+      needsControlledHeight: (widget: Widget) => {
+        const widgetVisualization = get(widget, 'config.visualization');
+        const flexibleHeightWidgets = [
```

Comment:
Do you know a better place for this collection of widgets types which do not need a controlled height?

Reply:
For now it's fine, because the plugin definitions contain the required knowledge. If we want to be able to delegate it, we should switch over to having a `needsControlledHeight` method on the `Widget` class. In that case we could delegate it further to the visualizations, in case of the `aggregation` widget.